### PR TITLE
Improve unique action and unload gun keybind

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Closes https://github.com/tgstation/TerraGov-Marine-Corps/issues/7241

New setting to control how your unique action will work

## Why It's Good For The Game

Less clunky to use

## Changelog
:cl:
qol: Unique action and unload gun works even if the gun is in your inactive hand
/:cl:

### CODE REVIEW

**Comment by KrabSpider** [comment]
...Theres already a keybind for unloading your gun.

File: code/modules/mob/mob_helpers.dm [hunk 1]
```diff
@@ -491,16 +491,18 @@ mob/proc/get_standard_bodytemperature()
...
 		return
 
 	var/obj/item/active_item = get_active_held_item()
-	if(!istype(active_item))
-		return
+	if(istype(active_item))
+		active_item.unique_action(src)
+	var/obj/item/inactive_item =get_inactive_held_item()
+	if(istype(inactive_item))
+		inactive_item.unique_action(src)
```
Comment (SplinterGP) [inline code/modules/mob/mob_helpers.dm:506]: Missing return

File: code/modules/projectiles/gun_helpers.dm [hunk 2]
```diff
@@ -972,11 +972,15 @@ should be alright.
 /// Signal handler to unload that gun if it's in our active hand
 /obj/item/weapon/gun/proc/unload_gun()
 	SIGNAL_HANDLER
-	if(gun_user?.get_active_held_item() != src)
-		return
 	unload(gun_user)
 	return COMSIG_KB_ACTIVATED
 
+/// Signal handler to unload that gun if it's in our active hand
```
Comment (SplinterGP) [inline code/modules/projectiles/gun_helpers.dm:978]: ?

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/modules/mob/mob_helpers.dm (hunk 1 above)
Comment (BraveMole) [reply code/modules/mob/mob_helpers.dm:506]: uh no that's intended, so it does exactly like the maro did

File: code/modules/projectiles/gun_helpers.dm (hunk 2 above)
Comment (BraveMole) [inline code/modules/projectiles/gun_helpers.dm:978]:
```suggestion
/// Signal handler to unload that gun if it's in our active hand
```
```suggestion
/// Signal handler to toggle the safety of the gun
```

File: code/modules/mob/mob_helpers.dm (hunk 1 above)
Comment (SplinterGP) [reply code/modules/mob/mob_helpers.dm:506]: What? So it will call unique action on BOTH of them at same time???

Comment (BraveMole) [reply code/modules/mob/mob_helpers.dm:506]: like the maro did, yes

Comment (SplinterGP) [reply code/modules/mob/mob_helpers.dm:506]: i dont know if it did but this is way too inconvinient

File: code/modules/mob/mob_helpers.dm [hunk 3]
```diff
@@ -491,16 +491,18 @@ mob/proc/get_standard_bodytemperature()
...
 	SIGNAL_HANDLER
 	. = COMSIG_KB_ACTIVATED //The return value must be a flag compatible with the signals triggering this.
 	if(incapacitated() || lying_angle)
 		return
 
 	var/obj/item/active_item = get_active_held_item()
-	if(!istype(active_item))
+	if(istype(active_item) && active_item.unique_action(src))
 		return
+	var/obj/item/inactive_item =get_inactive_held_item()
```
Comment (SplinterGP) [inline code/modules/mob/mob_helpers.dm:504]:
```suggestion
	var/obj/item/inactive_item = get_inactive_held_item()
```

**Review by SplinterGP — APPROVED** [review]

**Comment by Georgeur** [comment]
pls merge coder mans, I fucking need this, its so annoying to use revolvers without it

File: code/modules/mob/mob_helpers.dm [hunk 4]
```diff
@@ -491,16 +491,17 @@ mob/proc/get_standard_bodytemperature()
...
+/// Try to perform a unique action on the held items
 /mob/living/carbon/human/proc/do_unique_action()
 	SIGNAL_HANDLER
 	. = COMSIG_KB_ACTIVATED //The return value must be a flag compatible with the signals triggering this.
 	if(incapacitated() || lying_angle)
 		return
 
 	var/obj/item/active_item = get_active_held_item()
-	if(!istype(active_item))
+	if(istype(active_item) && active_item.unique_action(src))
```
Comment (SplinterGP) [inline code/modules/mob/mob_helpers.dm:502]: i don't know why this ?
```suggestion
	if(istype(active_item)
		active_item.unique_action(src)
```

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/modules/client/preferences_ui.dm [hunk 5]
```diff
@@ -578,7 +580,6 @@
 		if("fullscreen_mode")
 			fullscreen_mode = !fullscreen_mode
 			user.client?.set_fullscreen(fullscreen_mode)
-			return
```
Comment (BraveMole) [inline code/modules/client/preferences_ui.dm:581]: this was a rogue return

File: code/game/objects/items.dm [hunk 6]
```diff
@@ -357,7 +357,7 @@
 
 ///Anything unique the item can do, like pumping a shotgun, spin or whatever.
 /obj/item/proc/unique_action(mob/user)
-	return FALSE
+	return ITEM_HAS_NO_UNIQUE_ACTION
```
Comment (TiviPlus) [inline code/game/objects/items.dm:360]: Returns both bools and strings???

Comment (TiviPlus) [reply code/game/objects/items.dm:360]: Also also no KB handled return??

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Safety kb needs to be atomized out

**Review by TiviPlus — APPROVED** [review]

**Comment by SplinterGP** [comment]
Tivi approved so i assume this good, from what i reviewed before dinet noticw much and my pc broken now to review better

**Pull request merged**